src/lex.cpp

```diff
@@ -1157,8 +1160,12 @@ void lexer::skip_whitespace() {
 QLJS_WARNING_POP
 
 void lexer::skip_block_comment() {
```

Comment:
Blocking: Hmm. Let's instead create a separate `skip_html_comment` function. Three reasons:

* Performance. The extra checks for `-` inside the loop probably matter when parsing block comments. (I can check the performance impact if you want.)
* Readability. Block comment parsing is complicated because performance matters. I suspect HTML comments are rare, so performance is less important. We can implement less-optimized (thus probably more readable) code for HTML comments. For example, see what `skip_block_comment` looked like before commit fb2b951bf605dd928cf97d00832c5011f1858336: https://github.com/strager/quick-lint-js/blob/be4ef473ea1127617393fc038cd2da863f967849/src/lex.cpp#L762-L787
* Correctness. HTML comments have different rules than block comments (as you noticed). I suspect implementing these rules correctly in the same function as `skip_block_comment` will become really difficult. This is only a hunch, though; maybe you're implementing all the rules already.